internal/page/donor/upload_evidence.go

```diff
@@ -97,39 +102,61 @@ func readUploadEvidenceForm(r *http.Request) *uploadEvidenceForm {
 	}
 
 	// upload part
-	part, err = reader.NextPart()
-	if err != nil && err != io.EOF {
-		return &uploadEvidenceForm{Error: err}
-	}
+	var files []File
+	var formLevelError error
+	for {
+		part, err = reader.NextPart()
+		if err == io.EOF {
+			break
+		}
 
-	if part.FormName() != "upload" {
-		return &uploadEvidenceForm{Error: errors.New("unexpected field name")}
-	}
+		if err != nil {
+			formLevelError = err
+			break
+		}
 
-	buf := bufio.NewReader(part)
+		if part.FormName() != "upload" {
+			formLevelError = errors.New("unexpected field name")
+			break
+		}
 
-	sniff, _ := buf.Peek(peekSize)
-	if len(sniff) == 0 {
-		return &uploadEvidenceForm{Error: errUploadMissing}
-	}
+		buf := bufio.NewReader(part)
 
-	contentType := http.DetectContentType(sniff)
-	if contentType != "application/pdf" {
-		return &uploadEvidenceForm{Error: errUnexpectedContentType}
-	}
+		sniff, _ := buf.Peek(peekSize)
+		if len(sniff) == 0 {
+			formLevelError = errUploadMissing
+			break
+		}
 
-	var file bytes.Buffer
-	lmt := io.MultiReader(buf, io.LimitReader(part, maxUploadSize-peekSize+1))
+		contentType := http.DetectContentType(sniff)
+		if contentType != "application/pdf" {
+			files = append(files, File{Error: errUnexpectedContentType, Filename: part.FileName()})
+			continue
+		}
 
-	copied, err := io.Copy(&file, lmt)
-	if err != nil && err != io.EOF {
-		return &uploadEvidenceForm{Error: err}
-	}
-	if copied > maxUploadSize {
-		return &uploadEvidenceForm{Error: errUploadTooBig}
+		var f bytes.Buffer
+		lmt := io.MultiReader(buf, io.LimitReader(part, maxUploadSize-peekSize+1))
+
+		_, err := io.Copy(&f, lmt)
+
+		if err != nil {
+			file := File{Error: err, Filename: part.FileName()}
+
+			if errors.As(err, new(*http.MaxBytesError)) {
+				file.Error = errUploadTooBig
+			}
```

Comment:
@hawx - I'm not sure if I've missed something here but the check for `copied > maxUploadSize` could never be reached as Copy returns a `MaxBytesError` instead. I wasn't too sure if this was a per part or per form limit though?

Reply:
the limit in the `http.MaxBytesReader` call should be updated to `numberOfAllowedFiles*maxUploadSize+512`, then you'll be able to put back `copied > maxUploadSize` for each file. It is probably worth renaming `maxUploadSize` to `maxFileSize` to make more sense